v2/internal/test/fake/bundleendpoint.go

```diff
@@ -0,0 +1,165 @@
+package fake
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net"
+	"net/http"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/spiffe/go-spiffe/v2/bundle/spiffebundle"
+	"github.com/spiffe/go-spiffe/v2/internal/test"
+	"github.com/spiffe/go-spiffe/v2/internal/x509util"
+	"github.com/spiffe/go-spiffe/v2/spiffetls/tlsconfig"
+	"github.com/spiffe/go-spiffe/v2/svid/x509svid"
+	"github.com/stretchr/testify/assert"
+)
+
+type BundleEndpoint struct {
+	tb     testing.TB
+	wg     sync.WaitGroup
+	ctx    context.Context
+	port   int
+	server *http.Server
+	sm     *http.ServeMux
+	// Root certificates used by clients to verify server certificates.
+	rootCAs *x509.CertPool
+	// TLS configuration used by the server.
+	tlscfg *tls.Config
+	// SPIFFE bundle returned by this BundleEndpoint.
+	bundle *spiffebundle.Bundle
+}
+
+type BundleEndpointOption interface {
+	apply(*BundleEndpoint)
+}
+
+// BEOption groups the available options for a BundleEndpoint.
+var BEOption = new(bundleEndpointOption)
+
+func NewBundleEndpoint(ctx context.Context, tb testing.TB, port int, option ...BundleEndpointOption) *BundleEndpoint {
+	rootCAs, cert := test.CreateWebCredentials(tb)
+	tlscfg := &tls.Config{
+		Certificates: []tls.Certificate{*cert},
+	}
+
+	be := &BundleEndpoint{
+		tb:      tb,
+		wg:      sync.WaitGroup{},
+		ctx:     ctx,
+		port:    port,
+		sm:      http.NewServeMux(),
+		rootCAs: rootCAs,
+		tlscfg:  tlscfg,
+	}
+
+	for _, opt := range option {
+		opt.apply(be)
+	}
+
+	be.sm.HandleFunc("/healthcheck", be.healthcheck)
```

Comment:
If you take my suggestion and make a listener first, then the healthcheck stuff becomes unnecessary because the TLS port will have already been bound and the clients in your test will connect and be backlogged until the server starts accepting.

Reply:
This is what I wanted, just couldn't figure it out when tried to do it, so ended up doing the health-check thing, which I didn't like it...now I was able to do it (and am ashamed because it was so simple! :man_facepalming: ) and works like a charm, many thanks!